src/app/[locale]/(pages)/accedi/errore/page.tsx

```diff
@@ -99,10 +104,35 @@ const LoginErrorPage = () => {
   }, [errorCode]);
 
   const handleCancelBtn = () => {
-    if (isBrowser()) {
-      history?.back();
+    switch (getLoginFlow(prevLoginInfo)) {
+      case 'login_to_SessionExit':
+        pushWithLocale(ROUTES.LOGOUT_INIT);
+        break;
+      case 'login_to_Profile':
+        pushWithLocale(ROUTES.LOGIN);
+        break;
+      case 'login_to_UnlockAccessL3':
+        pushWithLocale(ROUTES.LOGIN_L3);
+        break;
+      default:
+        pushWithLocale(ROUTES.LOGIN);
+    }
```

Comment:
instead of using these values (login_to_SessionExit, login_to_Profile, login_to_UnlockAccessL3) use those you have collected in the variabille FLOW_PARAMS /src/app/[locale]/_utils/common.ts